python/nav/web/netmap/graph.py

```diff
@@ -162,20 +162,20 @@ def get_layer2_traffic(location_or_room_id=None):
             'netbox', 'to_netbox', 'to_interface__netbox'
         )
     else:
-        room = Room.objects.filter(id=location_or_room_id)
-        if room.exists():
-            location = Room.objects.get(id=location_or_room_id)
-        else:
-            location = Location.objects.get(id=location_or_room_id)
-            # Sanity check: Does the room exist?
-            # Fetch interfaces for devices in that room
         interfaces = Interface.objects.filter(
             to_netbox__isnull=False,
-            netbox__room__location=location
         ).select_related(
             'netbox', 'to_netbox', 'to_interface__netbox'
         )
 
+        try:
```

Comment:
Much nicer way to do it, yes.

Reply:
.. and I mean the entire try/except-block, setting a default interface queryset up front etc.